File: c++/sivitas_akademik.cpp
/*
    Saya Mohammad Ray Mosaid mengerjakan Latihan 2 dalam mata kuliah DPBO
    untuk keberkahanNya maka saya tidak melakukan kecurangan seperti yang telah dispesifikasikan.
    Aamiin.
*/

//memanggil header yang dibutuhkan
#include <iostream>
#include <string.h>

using namespace std;

//membuat class sivitas_akademik sebagai parent dari seluruh proses inheritance
class sivitas_akademik{
    //membuat atribut private
	private:
        string asal_universitas;
        string email_edu;

    public:
        //membuat konstruktor tanpa parameter
        sivitas_akademik(){
            this->asal_universitas = "";
            this->email_edu = "";
        }

        //membuat konstruktor dengan parameter
        sivitas_akademik(string asal_universitas, string email_edu){
            this->asal_universitas = asal_universitas;
            this->email_edu = email_edu;
        }

        //membuat fungsi untuk mengeset atribut private
        void set_asal_univ(string asal_universitas){
            this->asal_universitas = asal_universitas;
        }
        void set_email(string email_edu){
            this->email_edu = email_edu;
        }

        //membuat fungsi untuk me-return atribut private
        string get_asal_univ(){
            return asal_universitas;
        }
        string get_email(){
            return email_edu;
        }

        //membuat destruktor
        ~sivitas_akademik(){
        }
};

File: python/SivitasAkademik.py
#Saya Mohammad Ray Mosaid mengerjakan Latihan 2 dalam mata kuliah DPBO
#untuk keberkahanNya maka saya tidak melakukan kecurangan seperti yang telah dispesifikasikan.
#Aamiin.

#Memeanggil class SivitasAkademik
class SivitasAkademik:

    #variable/atribut private
    __asal_universitas = ""
    __email_edu = ""

    #membuat konsruktor untuk mengubah/mengakses variabel/atribut private
    def __init__(self, asal_universitas, email_edu):
        self.__asal_universitas = asal_universitas
        self.__email_edu = email_edu
    
    #membuat fungsi untuk mengeset variabel/atribut private
    def set_asal_universitas(self, asal_universitas):
        self.__asal_universitas = asal_universitas
    def set_email_edu(self, email_edu):
        self.__email_edu = email_edu
        
    #membuat fungsi untuk me-return hasil
    def get_asal_universitas(self):
        return self.__asal_universitas
    def get_email_edu(self):
        return self.__email_edu
    